cloudant-sync-datastore-android/src/test/java/com/cloudant/sync/replication/ReplicationServiceTest.java

```diff
@@ -459,4 +470,127 @@ else if (operationId == PeriodicReplicationService.COMMAND_STOP_REPLICATION && l
         }
     }
 
+    /**
+     * Check that when an intent is sent to the {@link PeriodicReplicationService} indicating that
+     * a replication should be started, the {@link PeriodicReplicationService} waits until
+     * {@link ReplicationService#setReplicators(Replicator[])} is called before processing
+     * the start replication operation.
+     */
+    public void testOnStartCommandStartReplicationWaitsForSetReplicators() {
+        PeriodicReplicationService service = new TestReplicationService(mMockContext);
+        Intent intent = new Intent(mMockContext, TestReplicationService.class);
+        intent.putExtra(PeriodicReplicationService.EXTRA_COMMAND, PeriodicReplicationService.COMMAND_START_REPLICATION);
+        service.setReplicationPolicyManager(mMockReplicationPolicyManager);
+
+        service.onCreate();
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        service.setOperationStartedListener(new PeriodicReplicationService
+            .OperationStartedListener() {
+            @Override
+            public void operationStarted(int operationId) {
+                if (operationId == PeriodicReplicationService.COMMAND_START_REPLICATION) {
+                    latch.countDown();
+                }
+            }
+        });
+
+        service.onStartCommand(intent, 0, 0);
+        try {
+            // Wait a bit to check the OperationStartedListener isn't called.
+            latch.await(1000, TimeUnit.MILLISECONDS);
+            assertEquals("CountDownLatch should not be decremented", latch.getCount(), 1);
+            service.setReplicators(mMockReplicators);
+            latch.await();
```

Comment:
I think this (and subsequent occurrences) would be better as something like:
 `assertTrue("The countdown should reach zero", latch.await(1, TimeUnit.MINUTES));`
That way at least if we end up on a failure path we aren't waiting indefinitely or relying on a global test timeout that could abort a build.